src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
```

Comment:
Using `Object.assign` here mutates `stateCopy`. Consider creating a new object to avoid mutation, such as `stateCopy = { ...stateCopy, ...action.extraData };`.